lib/SecretsCrypto/crypto.c

```diff
@@ -0,0 +1,1543 @@
+/*
```

Comment:
To avoid reinvention of the wheel in future I would suggest to use Glib2's GList and etc to work around with linked lists, but now it's Ok

Reply:
Same for GHashTable etc too, I think, yes.  My inexperience with glib (and frustrations getting the DBus GVariant marshalling/demarshalling working) led me to doing it this suboptimal way.

I guess one question is: should the glib types by used in the header / API, or only in the .c impl?  If the latter, we can fix this later as an improvement.